### ISSUE TITLE
Add current_user function

### ISSUE THREAD

**Issue by hegand** [prestodb/presto#6687]
It would be very nice if we can implement current_user function, which returns the current user's username.

Or I would like to ask if it is any way to get current username within a presto query?

**Comment by hegand** [comment]
I created a udf for that. The body of the function:

```
public class CustomStringFunctions {

    @ScalarFunction("current_user")
    @Description("Returns the current user")
    @SqlType(StandardTypes.VARCHAR)
    public static Slice current_user(ConnectorSession session)
    {
        return utf8Slice(session.getUser());
    }
}
```

**Comment by dain** [comment]
IIRC `CURRENT_USER` is a special construct in SQL like `CURRENT_DATE`, so adding it would require tweaking the grammar.  This change should be pretty straight forward.

**Comment by kokosing** [comment]
In my opinion even it is a special SQL construct it does not need be added to grammar, still it can be just a function. Although I don't have a strong opinion about this.

The thing I am concerned the most is that this function should evaluate to constant during planning. The same for CURRENT_DATE etc.

**Comment by dain** [comment]
The whole `session` is constant for the query.

**Comment by electrum** [comment]
This is specified in the SQL standard under `<general value specification>`. Thus, we should implement it using the special syntax.

**Comment by amitvc** [comment]
I am new to presto and would like to contribute by starting on beginner tasks. I can work on this ticket. My current experience with presto is I did a poc with 4 node cluster on EMR and with over half billion entries in multiple CSV files. Was able to perform some good tests with complex queries and played around with the system via cli and presto admin.

**Comment by wagnermarkd** [comment]
This can be closed since #10108 has been merged
